### PR TITLE
[MM-35059] Make /incident test bulk-data command create playbooks

### PR DESCRIPTION
**Pull request by arvinDarmawan**
Summary
This code will do two things : 
- create new command /incident test create-playbook  to create multiple playbooks
- update existing command /incident test bulk-data

Ticket Link
Jira card: https://mattermost.atlassian.net/browse/MM-35059
Github issue: https://github.com/mattermost/mattermost-server/issues/17469

```release-note
Release Note
- create new command /incident test create-playbook (parameter [total_playbooks]  to create multiple playbooks
- update existing command /incident test bulk-data
```

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Thank you for your contribution, @arvinDarmawan! :tada:

Could you update your branch with the latest changes in master? I think CI is failing because of that.

**Comment by arvinDarmawan** [comment]
> Thank you for your contribution, @arvinDarmawan! 🎉
> 
> Could you update your branch with the latest changes in master? I think CI is failing because of that.



> Thank you for your contribution, @arvinDarmawan! 🎉
> 
> Could you update your branch with the latest changes in master? I think CI is failing because of that.

ok, thank you

**Comment by arvinDarmawan** [comment]
Pull Request #17469

**Review by iomodo — APPROVED** [review]
Looks great! Thanks for the contribution @arvinDarmawan

File: server/command/command.go [hunk 1]
```diff
@@ -1062,6 +1068,39 @@ func (r *Runner) actionTest(args []string) {
...
+	}
+
+	numPlaybooks, err := strconv.Atoi(params[0])
+	if err != nil {
+		r.postCommandResponse("Error parsing the first argument. Must be a number.")
+		return
+	}
+
+	if numPlaybooks > 5 {
+		r.postCommandResponse("Cant generate playbooks since the number of playbook generated must be at most five")
```
Comment (agarciamontoro) [inline server/command/command.go:1084]: We can reword this to something more simple:

```suggestion
		r.postCommandResponse("Maximum number of playbooks is 5")
```

File: server/command/command.go [hunk 2]
```diff
@@ -1062,6 +1068,39 @@ func (r *Runner) actionTest(args []string) {
...
+		return
+	}
+
+	playbookIds := make([]string, 0, numPlaybooks)
+	for i := 0; i < numPlaybooks; i++ {
+		dummyPlaybook := dummyListPlaybooks[i]
+		dummyPlaybook.TeamID = r.args.TeamId
+		newPlaybookID, err := r.playbookService.Create(dummyPlaybook, r.args.UserId)
+		if err != nil {
+			r.warnUserAndLogErrorf("There was an error while creating playbook. Err: " + err.Error())
```
Comment (agarciamontoro) [inline server/command/command.go:1094]: You can use a formatted string here, as follows:
```suggestion
			r.warnUserAndLogErrorf("unable to create playbook: %v",  err)
```

File: server/command/command.go [hunk 3]
```diff
@@ -1062,6 +1068,39 @@ func (r *Runner) actionTest(args []string) {
...
+			return
+		}
+
+		playbookIds = append(playbookIds, newPlaybookID)
+	}
+
+	r.postCommandResponse(fmt.Sprintf("Playbooks successfully created : '%s'", playbookIds))
```
Comment (agarciamontoro) [inline server/command/command.go:1101]: This is good feedback, but we can make it even more helpful by directly linking to the playbooks. Something like the following (not tested, though):

```go
// We need this to construct the playbook URL
team, err := r.pluginAPI.Team.Get(r.args.TeamId)
if err != nil {
	r.warnUserAndLogErrorf("unable to retrieve team: %v", err)
	return
}

msg := "Playbooks successfully created"
for i, playbookID := range playbookIds {
	url := fmt.Sprintf("/%s/com.mattermost.plugin-incident-management/playbooks/%s", team.Name, playbookID)
	msg += fmt.Sprintf("\n- [%s](%s)", dummyListPlaybooks[i].Title, url)
}
```

File: server/command/command.go [hunk 4]
```diff
@@ -1247,6 +1286,280 @@ var incidentNames = []string{
 	"MM HA sync errors",
 }
 
+var dummyListPlaybooks = []playbook.Playbook{
+	{
+		Title:       "Blank Playbook",
+		Description: "This is example of empty playbook",
```
Comment (agarciamontoro) [inline server/command/command.go:1292]:
```suggestion
		Description: "This is an example of an empty playbook",
```

File: server/command/command.go [hunk 5]
```diff
@@ -1247,6 +1286,280 @@ var incidentNames = []string{
...
+					{
+						Title: "Create new tag in gitlab",
+					},
+				},
+			},
+		},
+	},
+	{
+		Title:       "Incident #4281",
+		Description: "There is error when access message from deleted channel",
```
Comment (agarciamontoro) [inline server/command/command.go:1412]:
```suggestion
		Description: "There is ab error when accessing message from deleted channel",
```

File: server/command/command.go [hunk 6]
```diff
@@ -1247,6 +1286,280 @@ var incidentNames = []string{
...
+					{
+						Title: "Test the result in production",
+					},
+				},
+			},
+		},
+	},
+	{
+		Title:       "Incident Collaboration Playbook",
+		Description: "### Describe the incident so that someone without prior knowledge can ramp up quickly. Two sentences is the ideal length.",
```
Comment (agarciamontoro) [inline server/command/command.go:1477]: We can come up with something here
```suggestion
		Description: "Sample playbook",
```

File: server/command/command.go [hunk 7]
```diff
@@ -1285,20 +1598,35 @@ func (r *Runner) generateTestData(numActiveIncidents, numEndedIncidents int, beg
...
+	var playbooks []playbook.Playbook
 	if len(playbooksResult.Items) == 0 {
-		r.postCommandResponse("You are not a member of any playbook. Create at least one playbook before generating the test data.")
-		return
-	}
+		dummyPlaybook := dummyListPlaybooks[rand.Intn(len(dummyListPlaybooks))]
+		dummyPlaybook.TeamID = r.args.TeamId
+		newPlayBookID, err := r.playbookService.Create(dummyPlaybook, r.args.UserId)
+		if err != nil {
+			r.warnUserAndLogErrorf("There was an error while creating playbook. Err: " + err.Error())
```
Comment (agarciamontoro) [inline server/command/command.go:1607]: Same suggestion here:
```suggestion
			r.warnUserAndLogErrorf("unable to create playbook: %v",  err)
```

File: server/command/command.go [hunk 8]
```diff
@@ -1285,20 +1598,35 @@ func (r *Runner) generateTestData(numActiveIncidents, numEndedIncidents int, beg
...
 	}
 
+	var playbooks []playbook.Playbook
 	if len(playbooksResult.Items) == 0 {
-		r.postCommandResponse("You are not a member of any playbook. Create at least one playbook before generating the test data.")
-		return
-	}
+		dummyPlaybook := dummyListPlaybooks[rand.Intn(len(dummyListPlaybooks))]
+		dummyPlaybook.TeamID = r.args.TeamId
+		newPlayBookID, err := r.playbookService.Create(dummyPlaybook, r.args.UserId)
```
Comment (agarciamontoro) [inline server/command/command.go:1605]: Nit: we usually write "book" in lowercase
```suggestion
		newPlaybookID, err := r.playbookService.Create(dummyPlaybook, r.args.UserId)
```

File: server/command/command.go [hunk 9]
```diff
@@ -1285,20 +1598,35 @@ func (r *Runner) generateTestData(numActiveIncidents, numEndedIncidents int, beg
...
+		if err != nil {
+			r.warnUserAndLogErrorf("There was an error while creating playbook. Err: " + err.Error())
+			return
+		}
 
-	playbooks := make([]playbook.Playbook, 0, len(playbooksResult.Items))
-	for _, thePlaybook := range playbooksResult.Items {
-		wholePlaybook, err := r.playbookService.Get(thePlaybook.ID)
+		newPlayBook, err := r.playbookService.Get(newPlayBookID)
```
Comment (agarciamontoro) [inline server/command/command.go:1611]: Same here
```suggestion
		newPlaybook, err := r.playbookService.Get(newPlaybookID)
```

File: server/command/command.go [hunk 10]
```diff
@@ -1285,20 +1598,35 @@ func (r *Runner) generateTestData(numActiveIncidents, numEndedIncidents int, beg
...
 		if err != nil {
 			r.warnUserAndLogErrorf("Error getting playbook: %v", err)
 			return
 		}
 
-		playbooks = append(playbooks, wholePlaybook)
+		playbooks = make([]playbook.Playbook, 0, 1)
+		playbooks = append(playbooks, newPlayBook)
```
Comment (agarciamontoro) [inline server/command/command.go:1618]: You can simply do this here:
```suggestion
		playbooks = []playbook.Playbook{newPlaybook}
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
This is really good, @arvinDarmawan, thank you! :tada: I left some suggestions below.

By the way, thank you for coming up with those sample playbooks, it's great!

**Comment by arvinDarmawan** [comment]
Hi @agarciamontoro , I already fix the revision

File: server/command/command.go [hunk 11]
```diff
@@ -1062,6 +1068,51 @@ func (r *Runner) actionTest(args []string) {
...
+		return
+	}
+
+	msg := "Playbooks successfully created"
+	for i, playbookID := range playbookIds {
+		url := fmt.Sprintf("/%s/com.mattermost.plugin-incident-management/playbooks/%s", team.Name, playbookID)
+		msg += fmt.Sprintf("\n- [%s](%s)", dummyListPlaybooks[i].Title, url)
+	}
+
+	r.postCommandResponse(fmt.Sprintf("Playbooks successfully created : '%s'", msg))
```
Comment (agarciamontoro) [inline server/command/command.go:1113]: We can just use `msg` here. Otherwise, the words "Playbooks successfully created" will be repeated:

```suggestion
	r.postCommandResponse(msg)
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Great job, @arvinDarmawan, thank you! Just a small change for the text displayed.

**Review by agarciamontoro — COMMENTED** [review]
Thanks! It looks like the lint check is failing. Could you take a look? https://app.circleci.com/pipelines/github/mattermost/mattermost-plugin-incident-collaboration/3284/workflows/9e06df12-8897-450a-856d-f45a8c385d83/jobs/18555 If you run `make check-style` locally, you'll get the error.

**Comment by arvinDarmawan** [comment]
> Thanks! It looks like the lint check is failing. Could you take a look? https://app.circleci.com/pipelines/github/mattermost/mattermost-plugin-incident-collaboration/3284/workflows/9e06df12-8897-450a-856d-f45a8c385d83/jobs/18555 If you run `make check-style` locally, you'll get the error.

Done @agarciamontoro

**Review by agarciamontoro — APPROVED** [review]
Great, @arvinDarmawan, thank you! :tada: Time for QA review :)

**Comment by lieut-data** [comment]
I've merged `origin/master` into this PR and tested it locally -- looks good! Will merge once build passes.

**Comment by lieut-data** [comment]
Thanks again, @arvinDarmawan!
